geopandas/io/tests/test_file.py

```diff
@@ -340,11 +340,26 @@ def test_read_file_filtered__rows_bbox__polygon(df_nybb):
     assert filtered_df_shape == (1, 5)
 
 
-def read_file_filtered_rows_invalid():
+def test_read_file_filtered_rows_invalid():
     with pytest.raises(TypeError):
         read_file(geopandas.datasets.get_path("nybb"), rows="not_a_slice")
 
 
+def test_read_file__columns():
+    gdf = geopandas.read_file(
+        geopandas.datasets.get_path("naturalearth_lowres"),
+        columns=["continent", "name"],
+    )
+    assert sorted(gdf.columns) == ["continent", "geometry", "name"]
```

Comment:
Since this is called in `from_features`
```
df = GeoDataFrame(rows, columns=columns)
```

is there any reason why `gdf.columns` should be other than the order passed in with `columns` parameter (plus `geometry` at the end or included in the list by the user)?  I think it would be good to test that you get back the order you asked for.

What should happen when the same column is specified multiple times?

Pandas would repeat them:
```
>>> rows
[{'a': 1}, {'a': 2}]
>>> DataFrame(rows, columns=['a', 'a'])
   a  a
0  1  1
1  2  2
```

I'm assuming the `columns` parameter here would produce the same behavior?

Reply:
Yep, it should - just added a test for that.